rpc-api/cmd/notify-rpc-api/grpc.go

```diff
@@ -0,0 +1,447 @@
+	var buf bytes.Buffer
+	_, err := io.Copy(&buf, reader)
+
+	return buf.Bytes(), err
+}
+
+func JSONDataNested(reader io.Reader) (json.RawMessage, error) {
+	var data DataJSON
+	err := json.NewDecoder(reader).Decode(&data)
+	return data.Data, err
+}
+
+func (c Client) Users() (json.RawMessage, error) {
+	resp, err := c.Get("/user")
+	if err != nil {
+		return nil, err
+	}
+
+	return JSONDataNested(resp.Body)
+}
+
+func (c Client) User(userID string) (json.RawMessage, error) {
+	resp, err := c.Get(fmt.Sprintf("/user/%s", userID))
+	if err != nil {
+		return nil, err
+	}
+
+	return JSONDataNested(resp.Body)
+}
+
+func (c Client) Service(serviceID string) (json.RawMessage, error) {
+	resp, err := c.Get(fmt.Sprintf("/service/%s", serviceID))
+	if err != nil {
+		return nil, err
+	}
+
+	return JSONDataNested(resp.Body)
+}
+
+func (c Client) Services(userID string) (json.RawMessage, error) {
+	resp, err := c.Get(fmt.Sprintf("/service?user_id=%s", userID))
+	if err != nil {
+		return nil, err
+	}
+
+	return JSONDataNested(resp.Body)
+}
+
+func (c Client) CreateTemplate(serviceID string, template Template) (json.RawMessage, error) {
+	var buf bytes.Buffer
+	err := json.NewEncoder(&buf).Encode(template)
+
+	resp, err := c.Post(
+		fmt.Sprintf("/service/%s/template", serviceID),
+		&buf,
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	return JSONDataNested(resp.Body)
+}
+
+func (c Client) Templates(serviceID string) (json.RawMessage, error) {
+	resp, err := c.Get(fmt.Sprintf("/service/%s/template", serviceID))
+	if err != nil {
+		return nil, err
+	}
+
+	return JSONDataNested(resp.Body)
+}
+
+type server struct {
+	c Client
+}
+
+type Service struct {
+	Active           bool     `json:"active"`
+	Branding         string   `json:"branding"`
+	CreatedBy        string   `json:"created_by"`
+	EmailFrom        string   `json:"email_from"`
+	Id               string   `json:"id"`
+	Name             string   `json:"name"`
+	OrganisationType string   `json:"organisation_type"`
+	RateLimit        int64    `json:"rate_limit"`
+	AnnualBilling    []string `json:"annual_billing"`
+	Permissions      []string `json:"permissions"`
+	UserToService    []string `json:"user_to_service"`
+	Users            []string `json:"users"`
+}
+
+func (service Service) Protoify() *notify.Service {
+	return &notify.Service{
+		Active:           service.Active,
+		Branding:         service.Branding,
+		CreatedBy:        service.CreatedBy,
+		EmailFrom:        service.EmailFrom,
+		Id:               service.Id,
+		Name:             service.Name,
+		OrganisationType: service.OrganisationType,
+		RateLimit:        service.RateLimit,
+		AnnualBilling:    service.AnnualBilling,
+		UserToService:    service.UserToService,
+		Permissions:      service.Permissions,
+		Users:            service.Users,
+	}
+
+}
+
+type Services []Service
+
+func (s Services) Protoify() *notify.Services {
+	services := [](*notify.Service){}
+
+	for _, service := range s {
```

Comment:
Per my comment above making the `notify` package import `pb`, this would now look like this:

```go
servicesp := [](*pb.Service){}

for _, s := range services {
  servicesp = append(servicesp, s.Protoify())
}
```

The reason for this is because looking at the method, it's unclear out of variables `s`, `services` and `service` what is proto and what is not. Suffixing with `p` solves this.